apps/api/src/app/user/dtos/user.dto.ts

```diff
@@ -0,0 +1,20 @@
+import {UserRole} from "../models/user-roles";
+import * as bcrypt from "bcrypt";
+
+const saltOfRounds : number=Math.random();
+export class UserDto {
+  id?: string;
+  name: string;
+  email: string;
+  roles: UserRole[];
+  password:string;
+  constructor(values:Partial<UserDto>){
+    if (values){
+      this.id=values.id;
+      this.email=values.email;
+      this.name=values.name;
+      this.roles=values.roles;
+      this.password=bcrypt.hash(values.password,saltOfRounds);
```

Comment:
**issue:** do not hash the password in the constructor (do it in the `service` or the `mapper`). You don't want everytime an userDto is created to have a hashed password